tests/csapi/thread_notifications_test.go

```diff
@@ -0,0 +1,196 @@
+package csapi_tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+	"github.com/matrix-org/complement/runtime"
+)
+
+func syncHasUnreadNotifs(roomID string, check func(gjson.Result, gjson.Result) bool) client.SyncCheckOpt {
```

Comment:
Might be worth a quick docstring (or whatever they're called in go) to describe what this does?

Something like this?

```
// Builds a `syncCheckOpt` which enforces that a sync result satisfies some `check` function 
// on the `unread_notifications` and `unread_thread_notifications` fields for the given room.
// The `unread_notifications` field must exist, or else the overall syncCheckOpt will be considered
// as failing.
```